### PR TITLE
[indexer] add ability to use struct type filter for get_owned_objects

### PR DESCRIPTION
**Pull request by emmazzz**
## Description 

This PR enables struct type and `MatchAny` struct type filters for get_owned_objects method.

## Test Plan 

Tested locally.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-indexer/src/indexer_reader.rs
Comment (gegaowp) [inline crates/sui-indexer/src/indexer_reader.rs:607]: add an index for object_type here?
https://github.com/MystenLabs/sui/blob/main/crates/sui-indexer/migrations_v2/2023-08-19-044023_objects/up.sql#L35

**Review by gegaowp — APPROVED** [review]
LGTM other than the minor indices comment.

File: crates/sui-indexer/src/indexer_reader.rs
Comment (bmwill) [inline crates/sui-indexer/src/indexer_reader.rs:580]: Can we double check that this string form is the same as is stored in the DB? I know we have some issues with canonical type format in the db

Comment (emmazzz) [reply crates/sui-indexer/src/indexer_reader.rs:580]: Yes I was actually using to_canonical_string initially but it didn't work while testing and changing it to to_string worked. But good point, I will double check the write side.

Comment (bmwill) [reply crates/sui-indexer/src/indexer_reader.rs:580]: oh the PR is still in flux: https://github.com/MystenLabs/sui/pull/14382

Comment (emmazzz) [reply crates/sui-indexer/src/indexer_reader.rs:580]: Will merge this one and follow up once that PR merges.

**Pull request merged**
